File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/product/ProductDetail.java
package vn.com.ltdt.Coffee_Shop.product;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t_product_detail")
public class ProductDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String Size;
    @NotNull(message = "The price cannot be null")
    private double price;
    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/domains/ProductDetail.kt
package com.ltdt.coffeeshop_android_native.data.domains

import com.ltdt.coffeeshop_android_native.data.remote.ProductDetailDTO

data class ProductDetail(
    val id: Int,
    val productId: Int,
    val size: String,
    val price: Double
)

fun ProductDetail.toProductDetailDTO(): ProductDetailDTO = ProductDetailDTO(
    id = id,
    productId = productId,
    size = size,
    price = price

)